File: ios/Classes/Channel/DataRequest.swift
//
//  Copyright (c) 2018 Loup Inc.
//  Licensed under Apache License v2.0
//

import Foundation

struct DataRequest: Codable, Equatable {
  let region: BeaconRegion
  let permission: Permission
  let inBackground: Bool
  
  static func ==(lhs: DataRequest, rhs: DataRequest) -> Bool {
    return (lhs.region.identifier == rhs.region.identifier)
  }
}

File: android/src/main/kotlin/io/intheloup/beacons/channel/DataRequest.kt
//  Copyright (c) 2018 Loup Inc.
//  Licensed under Apache License v2.0

package io.intheloup.beacons.channel

import io.intheloup.beacons.data.RegionModel
import io.intheloup.beacons.data.Permission

class DataRequest(
        val region: RegionModel,
        val permission: Permission,
        val inBackground: Boolean
)